selectors/views/custom_status.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {getCurrentUser, getUser} from 'mattermost-redux/selectors/entities/users';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {getMyPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
+import {Preferences} from 'mattermost-redux/constants';
+
+import {GlobalState} from 'types/store';
+
+export function getCustomStatus(state: GlobalState, userID?: string) {
+    const user = userID ? getUser(state, userID) : getCurrentUser(state);
+    const userProps = user.props || {};
+    return userProps.customStatus ? JSON.parse(userProps.customStatus) : {};
+}
+
+export function getRecentCustomStatuses(state: GlobalState) {
+    const preferences = getMyPreferences(state);
+    const key = getPreferenceKey(Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_RECENT_CUSTOM_STATUSES);
+    return preferences[key] && preferences[key].value ? JSON.parse(`${preferences[key].value}`) : [];
```

Comment:
Same thing here where we don't want to return a new empty array each time. This doesn't need a selector factory though since it has no arguments, so you just need something like
```javascript
export const getRecentCustomStatuses = createSelector(
    (state) => get(Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_RECENT_CUSTOM_STATUSES),
    (value) => {
        return value ? JSON.parse(value) : [];
    },
);
```